sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs

```diff
@@ -5,6 +5,24 @@ public partial class AuthenticationFailedException : System.Exception
         public AuthenticationFailedException(string message) { }
         public AuthenticationFailedException(string message, System.Exception innerException) { }
     }
+    public partial class AuthenticationRecord
+    {
+        internal AuthenticationRecord() { }
+        public string Authority { get { throw null; } }
+        public string HomeAccountId { get { throw null; } }
+        public string TenantId { get { throw null; } }
+        public string Username { get { throw null; } }
+        public static Azure.Identity.AuthenticationRecord Deserialize(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static System.Threading.Tasks.Task<Azure.Identity.AuthenticationRecord> DeserializeAsync(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public void Serialize(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { }
+        public System.Threading.Tasks.Task SerializeAsync(System.IO.Stream stream, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+    }
+    public partial class AuthenticationRequiredException : Azure.Identity.CredentialUnavailableException
```

Comment:
Usually exception names are saying what went wrong, not how to make it right.

Reply:
I'm going to stay with this name for this preview as it aligns with the other languages adding these features. We can discuss a possible rename of this exception type in the next preview.

Reply:
FWIW, I don't interpret this name as how to make it right. It's simply saying authentication is required. Yes, that strongly implies they need to authenticate, but `UnauthorizedException` in the BCL also strongly implies they need to modify (if possible) access controls, `NotImplementedException` strongly implies they (or someone) needs to implement, if desirable, etc.